api/src/zenbox_smt/bitvector_smt.rs

```diff
@@ -0,0 +1,192 @@
+////////////BITVECTOR SPARSE MERKLE TREE
+
+use primitives::{BigMerkleTree, Coord};
+use crate::ginger_calls::{GingerMerkleTreeParameters, Error, FieldElement, GingerMHTPath, GingerSMT, restore_ginger_smt, new_ginger_smt};
+use algebra::fields::mnt6753::FqParameters;
+use algebra::{FpParameters, ToBits};
+use std::path::Path;
+
+pub type BitvectorSMT = BigMerkleTree<GingerMerkleTreeParameters>;
+
+// Computes floor(log2(x))
+const fn log2(x: u64) -> usize {
```

Comment:
We already have this function in GingerLib, in algebra/mod.rs

Reply:
Thanks, then I will use it from Ginger!